application/configuration/types_copy_gen.go

```diff
@@ -0,0 +1,35 @@
+// generated by deep-copy --type config -o ./types_copy_gen.go ./types.go; DO NOT EDIT.
+
+package configuration
+
+// DeepCopy generates a deep copy of config
+func (o config) DeepCopy() config {
+	var cp config = o
+	if o.Kafka.Brokers != nil {
+		cp.Kafka.Brokers = make([]string, len(o.Kafka.Brokers))
+		copy(cp.Kafka.Brokers, o.Kafka.Brokers)
+	}
+	if o.Kafka.TopicDetail.ReplicaAssignment != nil {
```

Comment:
`ReplicaAssignment` and `ConfigEntries` is not required in the configuration structure. So I would recommend removing and just have `NumPartitions` and `NumPartitions`